src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2220,6 +2221,19 @@ static int32_t TryChangeSocketEventRegistrationInner(
                0,
                0,
                GetKeventUdata(data));
+#if defined(__FreeBSD__)
+        // FreeBSD seems to have some issue when setting read/write events together.
+        // As a workaround use separate kevent() calls.
```

Comment:
@wfurt, is there an issue number for the investigation and that we could include here?